### DISCUSSION TITLE
Adding Data To Cosmos - is there a fast way?

### DISCUSSION THREAD

**Discussion by situssoft** [Gremlinq/ExRam.Gremlinq#807] [Q&A] [answered]
What is the most efficient (fastest) way to add data to a Cosmos Graph.

I have a graph with approx 2K nodes and 3K edges, using individual awaited actions (AddV, AddE) in a loop seems very slow

Is there a beeter way to add this data?

Thanks

**Comment by ToddShelton** [comment, marked as answer]
[Microsoft's Bulk Executor library](https://learn.microsoft.com/en-us/azure/cosmos-db/bulk-executor-overview) works well.  It's fast and pretty easy to configure.  The quick start should get you going.

**Reply by situssoft** [reply]
Thanks for that, exactly what I need.
